uaclient/entitlements/fips.py

```diff
@@ -43,6 +44,25 @@ class FIPSCommonEntitlement(repo.RepoEntitlement):
 
     help_doc_url = "https://ubuntu.com/security/certifications#fips"
 
+    def _check_for_xenial_cloud_instances(self):
+        cfg_allow_xenial_fips_on_cloud = util.is_config_value_true(
+            config=self.cfg.cfg,
+            path_to_value="features.allow_xenial_fips_on_cloud",
+        )
+
+        if cfg_allow_xenial_fips_on_cloud:
+            return False
+
+        cloud_id = get_cloud_type()
+
+        if cloud_id and cloud_id.lower() in ("azure", "gce"):
+            series = util.get_platform_info()["series"]
+
+            if series.lower() == "xenial":
+                return True
```

Comment:
Let's pull this series check out and perform that first for two reasons:
- the series check is simple and fast and doesn't rely on cloud-init differences on trusty (no cloud-id cmd on trusty)
- I think it's more likely we'll exit on seeing Bionic instead of Xenial than seeing azure/gce vs AWS.